src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,563 @@
+
+    @Test
+    public void buildChangeValueUpdateAction_WithSameAbsoluteValues_ShouldNotBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final CartDiscountValue values =
+            CartDiscountValue.ofAbsolute(asList(MoneyImpl.of(10, EUR), MoneyImpl.of(10, USD)));
+        when(newCartDiscountDraft.getValue()).thenReturn(values);
+
+        final Optional<UpdateAction<CartDiscount>> changeValueUpdateAction =
+            buildChangeValueUpdateAction(CART_DISCOUNT_WITH_ABSOLUTE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeValueUpdateAction).isNotPresent();
+    }
+
+    @Test
+    public void buildChangeValueUpdateAction_WithSameAbsoluteValuesWithDifferentOrder_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final CartDiscountValue sameValuesWithDifferentOrder =
+            CartDiscountValue.ofAbsolute(asList(MoneyImpl.of(10, USD), MoneyImpl.of(10, EUR)));
+        when(newCartDiscountDraft.getValue()).thenReturn(sameValuesWithDifferentOrder);
+
+        final Optional<UpdateAction<CartDiscount>> changeValueUpdateAction =
+            buildChangeValueUpdateAction(CART_DISCOUNT_WITH_ABSOLUTE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeValueUpdateAction).contains(ChangeValue.of(sameValuesWithDifferentOrder));
+    }
+
+    @Test
+    public void buildChangeValueUpdateAction_WithSameValues_ShouldNotBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final CartDiscountValue tenPercent = CartDiscountValue.ofRelative(1000);
+        when(newCartDiscountDraft.getValue()).thenReturn(tenPercent);
+
+        final Optional<UpdateAction<CartDiscount>> changeValueUpdateAction =
+            buildChangeValueUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeValueUpdateAction).isNotPresent();
+    }
+
+    @Test
+    public void buildChangeValueUpdateAction_WithNullValues_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getValue()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> changeValueUpdateAction =
+            buildChangeValueUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeValueUpdateAction).contains(ChangeValue.of(null));
+    }
+
+    @Test
+    public void buildChangeCartPredicateUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getCartPredicate()).thenReturn("1 = 1");
+
+        final Optional<UpdateAction<CartDiscount>> changeCartPredicateUpdateAction =
+            buildChangeCartPredicateUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
```

Comment:
Don't you think it's simpler when you can see right here both values that are under test/comparison right here, instead of having to go the json file under this constant to see how the cart predicate there looks like? 

This is what I mean when you see the values under test close to each other:

```java
final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
when(newCartDiscountDraft.getCartPredicate()).thenReturn("foo");

final CartDiscount oldCartDiscount = mock(CartDiscount.class);
when(oldCartDiscount.getCartPredicate()).thenReturn("bar");

final Optional<UpdateAction<CartDiscount>> changeCartPredicateUpdateAction =
buildChangeCartPredicateUpdateAction(oldCartDiscount, newCartDiscountDraft);

assertThat(changeCartPredicateUpdateAction).contains(ChangeCartPredicate.of("foo"));
```

WDYT?

Reply:
07723af